static/app/views/ddm/customMetricsEventData.tsx

```diff
@@ -0,0 +1,206 @@
+import {Fragment, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {LinkButton} from 'sentry/components/button';
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
+import KeyValueList from 'sentry/components/events/interfaces/keyValueList';
+import {
+  MetricsSummary,
+  MetricsSummaryItem,
+} from 'sentry/components/events/interfaces/spans/types';
+import Link from 'sentry/components/links/link';
+import Pill from 'sentry/components/pill';
+import Pills from 'sentry/components/pills';
+import TextOverflow from 'sentry/components/textOverflow';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {MRI, Organization} from 'sentry/types';
+import {
+  formatMetricUsingUnit,
+  getDdmUrl,
+  getDefaultMetricOp,
+  getReadableMetricType,
+  MetricDisplayType,
+} from 'sentry/utils/metrics';
+import {hasDDMFeature} from 'sentry/utils/metrics/features';
+import {formatMRI, parseMRI} from 'sentry/utils/metrics/mri';
+import useOrganization from 'sentry/utils/useOrganization';
+
+function flattenMetricsSummary(
+  metricsSummary: MetricsSummary
+): {item: MetricsSummaryItem; key: string; mri: MRI}[] {
+  return Object.entries(metricsSummary).flatMap(([mri, items]) =>
+    items.map((item, index) => ({item, mri, key: `${mri}${index}`}))
+  );
+}
+
+export function CustomMetricsEventData({
+  metricsSummary,
+}: {
+  metricsSummary: MetricsSummary;
+}) {
+  const organization = useOrganization();
+  const metricsSummaryEntries = flattenMetricsSummary(metricsSummary);
+
+  if (!hasDDMFeature(organization) || metricsSummaryEntries.length === 0) {
+    return null;
+  }
+
+  return (
+    <EventDataSection type="custom-metrics" title={t('Emitted Metrics')}>
+      {metricsSummaryEntries.map(({mri, item, key}) => {
+        return (
+          <Fragment key={key}>
+            <KeyValueList
+              shouldSort={false}
+              data={[
+                {
+                  key: 'name',
+                  subject: t('Name'),
+                  value: <TextOverflow>{formatMRI(mri)}</TextOverflow>,
+                  actionButton: (
+                    <LinkButton
+                      size="xs"
+                      to={getDdmUrl(organization.slug, {
+                        widgets: [
+                          {
+                            mri,
+                            displayType: MetricDisplayType.LINE,
+                            op: getDefaultMetricOp(mri),
+                          },
+                        ],
+                      })}
+                    >
+                      {t('Open in Metrics')}
+                    </LinkButton>
```

Comment:
Should this open the plain metric or also apply all of the tags? I'm not sure.

Reply:
Hmm, I think it makes sense 👍 
Small nit though: The current UI is not ideal for that as the Button is inside the name field and not visually associated with the rest of the metric summary. We will probably need to revisit that.